Emby.Server.Implementations/ScheduledTasks/Tasks/CleanupCollectionAndPlaylistPathsTask.cs

```diff
@@ -0,0 +1,147 @@
+    public CleanupCollectionAndPlaylistPathsTask(
+        ILocalizationManager localization,
+        ICollectionManager collectionManager,
+        IPlaylistManager playlistManager,
+        ILogger<CleanupCollectionAndPlaylistPathsTask> logger,
+        IProviderManager providerManager,
+        IFileSystem fileSystem)
+    {
+        _localization = localization;
+        _collectionManager = collectionManager;
+        _playlistManager = playlistManager;
+        _logger = logger;
+        _providerManager = providerManager;
+        _fileSystem = fileSystem;
+    }
+
+    /// <inheritdoc />
+    public string Name => _localization.GetLocalizedString("TaskCleanCollectionsAndPlaylists");
+
+    /// <inheritdoc />
+    public string Key => "CleanCollectionsAndPlaylists";
+
+    /// <inheritdoc />
+    public string Description => _localization.GetLocalizedString("TaskCleanCollectionsAndPlaylistsDescription");
+
+    /// <inheritdoc />
+    public string Category => _localization.GetLocalizedString("TasksMaintenanceCategory");
+
+    /// <inheritdoc />
+    public async Task ExecuteAsync(IProgress<double> progress, CancellationToken cancellationToken)
+    {
+        var collectionsFolder = await _collectionManager.GetCollectionsFolder(false).ConfigureAwait(false);
+        if (collectionsFolder is null)
+        {
+            _logger.LogDebug("There is no collections folder to be found");
+        }
+        else
+        {
+            var collections = collectionsFolder.Children.OfType<BoxSet>().ToArray();
+            _logger.LogDebug("Found {CollectionLength} boxsets", collections.Length);
+
+            for (var index = 0; index < collections.Length; index++)
+            {
+                var collection = collections[index];
+                _logger.LogDebug("Checking boxset {CollectionName}", collection.Name);
+
+                CleanupLinkedChildren(collection, cancellationToken);
+                progress.Report(50D / collections.Length * (index + 1));
+            }
+        }
+
+        var playlistsFolder = _playlistManager.GetPlaylistsFolder();
+        if (playlistsFolder is null)
+        {
+            _logger.LogDebug("There is no playlists folder to be found");
+            return;
+        }
+
+        var playlists = playlistsFolder.Children.OfType<Playlist>().ToArray();
+        _logger.LogDebug("Found {PlaylistLength} playlists", playlists.Length);
+
+        for (var index = 0; index < playlists.Length; index++)
+        {
+            var playlist = playlists[index];
+            _logger.LogDebug("Checking playlist {PlaylistName}", playlist.Name);
+
+            CleanupLinkedChildren(playlist, cancellationToken);
+            progress.Report(50D / playlists.Length * (index + 1));
+        }
+    }
+
+    private void CleanupLinkedChildren<T>(T folder, CancellationToken cancellationToken)
+        where T : Folder
+    {
+        List<LinkedChild> itemsToRemove = null;
```

Comment:
```suggestion
        List<LinkedChild>? itemsToRemove = null;
```
Should fix build error